src/main/java/uk/gov/justice/probation/courtcaseservice/controller/model/Defendant.java

```diff
@@ -55,4 +57,8 @@ public String getSex() {
     @Valid
     @NotEmpty
     private final List<OffenceRequestResponse> offences;
+
+    public String getPersonId(){
+        return Optional.ofNullable(personId).orElse(UUID.randomUUID().toString());
```

Comment:
should the personId be assigned with the generated uuid before returning? Unlikely but prone to generating new uuid for every getPersonId call.
Also not sure the DTO should take this responsibility. I'd generate it outside of the DTO when needed if dto.getPersionId is null. Not precious about it but having a separate @Component to do generate the personId helps unit testing as mocking UUID is non trivial

Reply:
As this is a temp solution until a proper person record service is built I have chosen this way as this will be easy to remove at a later stage when everything is ready.